dart/lib/src/event_processor/deduplication_event_processor.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:async';
+import '../event_processor.dart';
+import '../protocol.dart';
+import '../sentry_options.dart';
+
+const eventsToKeepForDeduplication = 20;
+
+/// Deduplicates events with the same [SentryEvent.throwable].
+/// It keeps track of the last [eventsToKeepForDeduplication]
+/// events. Older events aren't considered for deduplication.
+/// This is a hard
+/// Caveats:
+/// It does not work in the following case:
+/// ```dart
+/// var fooOne = Exception('foo');
+/// var fooTwo = Exception('foo');
+/// ```
+/// because (fooOne == fooTwo) equals false
+class DeduplicationEventProcessor extends EventProcessor {
+  DeduplicationEventProcessor(this._options);
+
+  // Map from exception to object
+  final Map<Object, Object?> exceptionMap = {};
```

Comment:
looks like we don't use the value at all, wondering if `Queue` or `Dequeue` would make more sense

Reply:
Yep, I wondered about that in the Java and C# implementation, too.

Reply:
https://github.com/getsentry/sentry-java/issues/1533 lets see if there was a reason

Reply:
I think it's just to make look ups fast. For maps and sets you can just do a 'contains' lookup. While you can do that for lists too, it's not O(1) but O(n) in the worst case. At least if I remember the big O notation correctly 😅

Reply:
yep, that's a good point, let's see if it's the case :D

Reply:
the reason was that Java and .NET don't have a List interface of weak references, but only a `WeakHashMap`, so when the GC is doing its job, it'd clean up the data automatically from the HashMap, also, because of lookup time.
to improve lookup time, we can use a data structure that is sorted or O(1), but instead of the object itself, we insert and check the hashCode of the object.
I believe Java could be also improved, see https://github.com/getsentry/sentry-java/issues/1533#issuecomment-863016739